### PR TITLE
fix: add missing continue to shard cache put

### PR DESCRIPTION
**Pull request by Longarithm**
If we are unable to remove element with hash taken from deletions queue, we remove LRU item from cache. This is not optimal - we should continue iterating over deletions queue until we don't exhaust it.

## Testing

Existing tests - but we will have to add more https://github.com/near/nearcore/issues/7498

### CODE REVIEW

File: core/store/src/trie/trie_storage.rs [hunk 1]
```diff
@@ -144,6 +144,7 @@ impl TrieCacheInner {
                     }
                     None => {
                         self.metrics.shard_cache_pop_misses.inc();
+                        continue;
```
Comment (matklad) [inline core/store/src/trie/trie_storage.rs:147]: Good catch! If we already caught a bug here, let's try to compress this experssion to make it simpler and less error prone. Something like this should work I think: 

```rust
            // First, try to evict value using the key from deletions queue.
            if let Some(key) = self.deletions.pop() {
                match self.cache.pop(&key) {
                    Some(value) => {
                        self.total_size -= value.len() as u64;
                        self.metrics.shard_cache_pop_hits.inc();
                    }
                    None => self.metrics.shard_cache_pop_misses.inc(),
                }
                continue;
            }

```

**Review by matklad — APPROVED** [review]
Do we have some existing tests which cover this scenario? Can we extend them to include assertions about the state of the cache?

File: core/store/src/trie/trie_storage.rs (hunk 1 above)
Comment (mm-near) [reply core/store/src/trie/trie_storage.rs:147]: +1 for adding the test (as we shouldn't be in a hurry now).

**Review by mm-near — CHANGES_REQUESTED** [review]
please add the test.

**Comment by akhi3030** [comment]
I suspect this PR is no longer being actively worked on.  Feel free to reopen when you are ready to start working on it again.

**Pull request closed without merging**